### PR TITLE
Handle target in OpenInApp response

### PR DESCRIPTION
**Pull request by glpatcern**
This PR adds the `OpenInApp.target` and `AppProviderInfo.action` properties to the respective responses (from `/app/open` and `/app/list`), to support different app integrations. Builds on top of https://github.com/cs3org/cs3apis/pull/212.

In addition, the archiver was extended to use the name of the file/folder as opposed to `download`, and to include a query parameter to override the archive type, as it will be used in an upcoming app, see #4084.

### CODE REVIEW

**Comment by labkode** [comment]
Rebase needed

File: internal/http/services/archiver/handler.go [hunk 1]
```diff
@@ -229,13 +230,27 @@ func (s *svc) Handler() http.Handler {
 			return
 		}
 
-		userAgent := ua.Parse(r.Header.Get("User-Agent"))
+		archTypeParams, ok := v["arch_type"] // optional, either "tar" or "zip"
```
Comment (gmgigi96) [inline internal/http/services/archiver/handler.go:233]: Use https://pkg.go.dev/net/url#Values.Get to simplify the code

Comment (glpatcern) [reply internal/http/services/archiver/handler.go:233]: OK!

File: internal/http/services/appprovider/appprovider.go [hunk 2]
```diff
@@ -512,3 +522,14 @@ func resolveViewMode(res *provider.ResourceInfo, vm string) gateway.OpenInAppReq
 	}
 	return viewMode
 }
+
+func appTargetToString(t apppb.Target) string {
```
Comment (labkode) [inline internal/http/services/appprovider/appprovider.go:526]: In case the the target is `invalid` you'll be opening it with `iframe`?

Comment (glpatcern) [reply internal/http/services/appprovider/appprovider.go:526]: Yeah, I thought about that, after all any future app provider would have to specify the target so let's not assume a default: I return empty string now (i.e. likely failing on the frontend)

Comment (glpatcern) [reply internal/http/services/appprovider/appprovider.go:526]: Not sure it makes much sense given the context (I'd have preferred a default value in protobuf, but it's an enum so it must have a value). But I don't have a strong opinion, so ok I changed it to `invalid`.

**Review by labkode — APPROVED** [review]

**Comment by labkode** [comment]
Breaking change:
Default changes from "download_archive" http prefix to "archiver".

**Pull request merged**